dev-guide/kubernetes-continuous-rebase.md

```diff
@@ -0,0 +1,153 @@
+2. We receive early signals if an upstream change breaks OCP, enabling
+   us to address the issue promptly either in the upstream code or on
+   our side.
+3. The rebase PR should be ready to be landed as soon as the upstream
+   code becomes generally available (GA).
+
+To implement this proposal, the following steps are required:
+
+### Watcher
+
+For each OCP release, we will designate a watcher to participate in
+the process. Ideally, it should be the same person who will execute
+the final rebase.
+
+A watcher is responsible for ensuring that the remaining steps
+outlined below are executed without errors.
+
+Although some manual work is required, it should not occupy their
+entire daily working time.
+
+### A -next branch
+
+For each of the dependencies listed below, a new branch called
+`ocp-next` is created with their Kubernetes dependencies updated:
+
+* openshift/api
+* openshift/client-go
+* openshift/library-go
+* openshift/apiserver-library-go
+
+Initially, this can be done manually on a weekly basis. In the future,
+certain parts of this process can potentially be automated, requiring
+manual intervention only when the automation fails.
+
+This process should already uncover some future issues, requiring
+fixes on unit tests or Makefiles for instance.
+
+### CI Job
+
+The goal of the CI job is to detect if our downstream patches create
+any code conflicts when applied to the upstream code. In addition to
+that, it will uncover potential issues with dependencies and
+generated code.
+
+In short, the new CI job will:
+
+1. Take a series of downstream patches and apply them against the
+   upstream code.
+2. Pin the dependencies mentioned above to the HEAD of their
+   respective `ocp-next` branches.
+3. Update the auto-generated code and docs (i.e., `make update`).
+4. Make sure the codebase is in a sane state by executing automated
+   verification and testing with `make` (i.e., `test`, `verify`,
+   `build`, etc.).
+5. Commit and push the local changes to an `ocp-next` branch in a
+   remote repository.
+6. Update or create the Pull Request.
+
+If the job fails to execute any of the steps above, the watcher is
+responsible for fixing whatever is preventing the job from
+succeeding. Examples of fixes include:
+
+1. Making a code change to the downstream patch to address a code
+   conflict.
+2. Creating an upstream PR to correct any breaking change.
+3. Creating a new downstream patch to rectify an incorrect assumption
+   in our operators.
+
+A prototype of this workflow is available
+[here](https://github.com/bertinatto/ocp-next/blob/master/do.sh).
+
+### Open Questions
+
+This proposal assumes that all downstream patches are located in a
+specific directory, such as the `patches` directory in [this
+prototype](https://github.com/bertinatto/ocp-next/tree/master/patches).
+
+However, it is unclear how we can ensure that this directory remains
+up-to-date with the latest patches imported into our
+openshift/kubernetes fork.
```

Comment:
See https://github.com/soltysh/rebase which does a log of currently applied patches, w/o the necessity to store them anywhere. It's still possible we'll miss a patch or two, but I'm willing to accept that initially.

Reply:
The tooling is mentioned right below.